File: app01_react/backend/src/main/java/com/securevision/dto/FrameworkResponse.java
package com.securevision.dto;

import com.securevision.entity.Framework;
import lombok.Builder;

@Builder
public record FrameworkResponse(
        String id,
        String code,
        String name,
        String version,
        String description,
        String referenceUrl
) {
    public static FrameworkResponse from(Framework framework) {
        return FrameworkResponse.builder()
                .id(framework.getId().toString())
                .code(framework.getCode())
                .name(framework.getName())
                .version(framework.getVersion())
                .description(framework.getDescription())
                .referenceUrl(framework.getReferenceUrl())
                .build();
    }
}

File: app10_csharp_react/backend/src/SharpGuard.Api/Dtos/FrameworkResponse.cs
using SharpGuard.Api.Entities;

namespace SharpGuard.Api.Dtos;

public record FrameworkResponse(
    string Id,
    string Code,
    string Name,
    string Version,
    string? Description,
    string? ReferenceUrl)
{
    public static FrameworkResponse From(Framework framework) => new(
        framework.Id.ToString(),
        framework.Code,
        framework.Name,
        framework.Version,
        framework.Description,
        framework.ReferenceUrl);
}
